dotCMS/src/integration-test/java/com/dotcms/junit/RuleWatcher.java

```diff
@@ -14,6 +34,100 @@ protected void starting(Description description) {
+                    Logger.info(MainBaseSuite.class,
+                            "Indexer is not empty, waiting for it to finish");
+                    if (!ReindexThread.isWorking()) {
+                        Logger.info(MainBaseSuite.class, "Indexer was not running, unpausing now");
+                        ReindexThread.unpause();
+                    }
+                    TestDataUtils.waitForEmptyQueue();
+                    boolean queueEmpty = TestDataUtils.waitForEmptyQueue();
+                    Logger.info(MainBaseSuite.class, "Indexer Complete=" + queueEmpty);
+                }
+            } catch (DotDataException e) {
+                throw new RuntimeException("Error accessing Index", e);
+            }
+        }
+    }
+
+
+    private void configOverrideTrackerBefore(Description description) {
+        if (Config.getBooleanProperty(USE_CONFIG_TEST_OVERRIDE_TRACKER, false)) {
+            if (Logger.isDebugEnabled(RuleWatcher.class)) {
+                Config.getOverrides().forEach((k, v) -> Logger.warn(this.getClass(),
+                        () -> "Config overrides before test: " + k + " = " + v));
+            }
+            overrides.put(description.hashCode(), Config.getOverrides());
+        }
+    }
+
+    private void configOverrideTrackerAfter(Description description) {
+        if (Config.getBooleanProperty(USE_CONFIG_TEST_OVERRIDE_TRACKER, false)) {
+            Logger.warn(RuleWatcher.class, "Checking for modified Config overrides...");
+            Map<String, String> modifiedOverrides = Config.compareOverrides(
+                    overrides.get(description.hashCode()));
+            overrides.remove(description.hashCode());
+            if (!modifiedOverrides.isEmpty()) {
+                String mapContents = modifiedOverrides.entrySet().stream()
+                        .map(entry -> entry.getKey() + "=" + entry.getValue())
+                        .reduce((a, b) -> a + ", " + b).orElse("empty");
+                Logger.warn(IntegrationTestBase.class,
+                        "Modified Config overrides after:" + mapContents);
+            }
+        }
+    }
+
+    private static void transactionTrackerAfter(Description description) {
+        if (Config.getBooleanProperty(USE_TEST_TRANSACTION_TRACKER, false)) {
+            if (DbConnectionFactory.inTransaction()) {
+                Logger.error(RuleWatcher.class,
+                        "Test " + description + " has open transaction after");
+                Object responseAction = Config.getStringProperty(
+                        TEST_TRANSACTION_TRACKER_AUTO_CLOSE_TYPE, TRANSACTION_AUTO_CLOSE_DEFAULT);
+                try {
+                    if (Config.getStringProperty(TEST_TRANSACTION_TRACKER_AUTO_CLOSE_TYPE,
+                            TRANSACTION_AUTO_CLOSE_DEFAULT).equals("commit")) {
+                        HibernateUtil.commitTransaction();
+                    } else {
+                        HibernateUtil.rollbackTransaction();
```

Comment:
This else doesn't seem to be required here as you always rollback the transaction at the end

Reply:
Call to second rollback should not be there, fixed